layerindex/tools/github-fetch.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+# Github fetch utility
+
+# Copyright (C) 2017 Intel Corporation
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os
+import argparse
+import json
+import requests
+import re
+import subprocess
+
+def fetchall(args):
+    link_re = re.compile('<(http[^>]+)>; rel="([a-zA-Z0-9]+)"')
+
+    keys = {'orgname': args.organisation,
+            'access_token': args.access_token,
+            'per_page': 100
+            }
+    url = 'https://api.github.com/orgs/{orgname}/repos?access_token={access_token}&per_page={per_page}'
+    url = url.format(**keys)
+
+    existing = os.listdir(args.outdir)
+    for name in existing:
+        if name.endswith('.deleted'):
+            print('Directories marked deleted (suffix .deleted) still exist in output path - remove these to continue')
+            sys.exit(1)
+
+    if args.resume_from:
+        fetching = False
+    else:
+        fetching = True
+
+    failed = []
+    repos = None
+    while True:
+        session = requests.Session()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *session*. The resource is allocated by call *sessions.Session*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *session* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).